cmd/client-s3.go

```diff
@@ -467,6 +467,39 @@ func (c *s3Client) Get() (io.Reader, *probe.Error) {
 	return reader, nil
 }
 
+// Get - get object with metadata.
+func (c *s3Client) GetWithMetadata() (io.Reader, map[string][]string, *probe.Error) {
+	bucket, object := c.url2BucketAndObject()
+	reader, e := c.api.GetObject(bucket, object)
+	if e != nil {
+		errResponse := minio.ToErrorResponse(e)
+		if errResponse.Code == "AccessDenied" {
+			return nil, nil, probe.NewError(PathInsufficientPermission{Path: c.targetURL.String()})
+		}
+		if errResponse.Code == "NoSuchBucket" {
+			return nil, nil, probe.NewError(BucketDoesNotExist{
+				Bucket: bucket,
+			})
+		}
+		if errResponse.Code == "InvalidBucketName" {
+			return nil, nil, probe.NewError(BucketInvalid{
+				Bucket: bucket,
+			})
+		}
+		if errResponse.Code == "NoSuchKey" || errResponse.Code == "InvalidArgument" {
+			return nil, nil, probe.NewError(ObjectMissing{})
+		}
+		return nil, nil, probe.NewError(e)
+	}
+	objInfo, e := reader.Stat()
+	if e != nil {
+		return nil, nil, probe.NewError(e)
+	}
+	metadata := objInfo.Metadata
+	metadata.Set("Content-Type", objInfo.ContentType)
```

Comment:
We have to since Put needs it, we are kind of extracting the ContentType field back into metadata for consumption by caller.